### DISCUSSION TITLE
How can I use APIs in dds::core::cond::StatusCondition? [13601]

### DISCUSSION THREAD

**Discussion by weithegreat** [eProsima/Fast-DDS#2439] [Q&A]
Hey folks, I'm new to fastDDS. I saw that we have both include/dds and include/fastdds/ however include/dds  directory is not installed when I install fastDDS.
There;re APIs in include/dds/core/cond/Condition.hpp I want to use, like handler, but it's not in include/fastdds/dds/core/condition/Condition.hpp

I was wondering why do we have these two directories? Is include/dds/core/cond/Condition.hpp  not supposed to be used?

**Comment by JLBuenoLopez** [comment]
Sorry for the late reply, @weigewansui,

The API included in the `include/dds` API is not installed because it is still under development. The [OMG DDS Specification](https://www.omg.org/spec/DDS/1.4/About-DDS/) defines a PIM (Platform Independent Model) and a PSM (Platform Specific Model) APIs. Currently, Fast DDS only has implemented the PIM API and the PSM API (under the `dds` folder) is still being developed and thus it is not possible to use it yet.

However, you are probably interested on using the StatusCondition API and I am glad to say that this feature is available in the PIM API (the one already available to the users) since release 2.4.0 (last September 2021)!
